### ISSUE TITLE
no-nested-function-param-reassign

### ISSUE THREAD

**Issue by niftylettuce** [eslint/eslint#11915]
It would be nice and have saved me a lot of time earlier if there was a rule that prevented re-assignment of parameters for nested functions.

The example below shows where I re-defined `args` via a nested function, it would be nice to have a rule that disallows this and errors against it:

```js
function foo(...args) {
  const obj = {};
  ['red','green','blue'].forEach(color => {
    obj[color] = (...args) => {  // <--- throw a warning/error here
      console.log('color', color);
      console.log(...args);
    };
  });
}
```

**Comment by kaicataldo** [comment]
Thanks for the issue. Have you tried the [no-shadow](https://eslint.org/docs/rules/no-shadow) rule?
